components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -1414,4 +1418,50 @@ private org.wso2.carbon.user.core.common.User getUserFromUserID(String userId) {
                     ErrorMessage.ERROR_RETRIEVING_USERSTORE_MANAGER.getDescription());
         }
     }
+
+    /**
+     * This method will retrieve the user registrants configured for the given application id by calling the backend
+     * services.
+     *
+     * @param applicationId The application id
+     * @return the user registrants list object
+     */
+    public UserRegistrantsList getConfiguredUserRegistrants(String applicationId) {
+
+        String tenantDomain = ContextLoader.getTenantDomainFromContext();
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Retrieving the user registrants configured for the application," + applicationId);
+            }
+            List<AuthAttributeHolder> availableAuthAttributeHolders =
```

Comment:
Why do we call this type `AuthAttributeHolder` ?. How are these related to registrants ?

Reply:
This is a new component added to the governance which will validate the attributes required by each of the signup-supported authenticators. They will only tell the required attributes and verifies whether these attributes are given when the user is onboarding with that specific option